t/python/t0007-watchers.py

```diff
@@ -76,8 +77,70 @@ def test_msg_watcher_bytes(self):
             self.assertIsNotNone(mw)
 
 
+class TestSignal(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.f = flux.Flux()
+
+    def test_signal_watcher_invalid(self):
+        """Add an invalid signal"""
+        with self.assertRaises(EnvironmentError):
+            self.f.signal_watcher_create(
+                -500, lambda x, y: x.fatal_error("signal should not fire")
+            )
+        with self.assertRaises(EnvironmentError):
+            self.f.signal_watcher_create(
+                0, lambda x, y: x.fatal_error("signal should not fire")
+            )
+        with self.assertRaises(EnvironmentError):
+            self.f.signal_watcher_create(
+                500, lambda x, y: x.fatal_error("signal should not fire")
+            )
+
+    def test_s0_signal_watcher_add(self):
+        """Add a signal watcher"""
+        with self.f.signal_watcher_create(
+            2, lambda x, y, z, w: x.fatal_error("signal should not fire")
+        ) as sigw:
+            self.assertIsNotNone(sigw)
+
+    def test_s1_signal_watcher_remove(self):
+        """Remove a timer"""
+        sigw = self.f.signal_watcher_create(
+            2, lambda x, y: x.fatal_error("signal should not fire")
+        )
+        sigw.stop()
+        sigw.destroy()
+
+    def test_signal_watcher(self):
+        cb_called = [False]
+
+        def cb(handle, watcher, signum, args):
+            cb_called[0] = True
+            handle.reactor_stop()
+
+        def raise_signal(handle, wathcer, revents, args):
+            os.kill(os.getpid(), signal.SIGUSR1)
+
+        def stop(h, w, r, y):
+            h.reactor_stop()
+
+        with self.f.signal_watcher_create(signal.SIGUSR1, cb) as watcher:
+            self.assertIsNotNone(watcher, msg="signalWatcher create")
+            to = self.f.timer_watcher_create(0.10, raise_signal)
```

Comment:
minor nit, when there are two things, I think it should be named `to1` and `to2`, instead of `to` and `to2` :P

Reply:
heh, okay...